tests/karate/src/test/java/be/model/Transaction.java

```diff
@@ -0,0 +1,211 @@
+    this.lockTime = 0;
+  }
+
+  /**
+   * Simplified version of transaction computations in DigitalCashViewModel in android frontend.
+   * This only works for one transaction and does not keep track of previous transactions!
+   *
+   * @return a transaction with updated input and output values
+   * @throws GeneralSecurityException
+   */
+  public void issueInitialCoins(String receiverPublicKey, String senderPublicKey, String senderPrivateKey, long amountToGive)
+    throws GeneralSecurityException {
+    Output output = new Output(
+      amountToGive,
+      new ScriptOutput(type, Hash.hash(receiverPublicKey.getBytes(StandardCharsets.UTF_8))));
+    outputs.add(output);
+
+    Map<String, Integer> inputsPairs = new HashMap<>();
+    inputsPairs.put(tx_out_hash_coinbase, 0);
+
+    byte[] toSign = Transaction.computeSigOutputsPairTxOutHashAndIndex(outputs, inputsPairs)
+      .getBytes(StandardCharsets.UTF_8);
+
+    // Create a signature for the transaction using the sender's private key
+    PublicKeySign publicKeySign = new Ed25519Sign(Base64Utils.decode(senderPrivateKey));
+    byte[] signBytes = publicKeySign.sign(toSign);
+    String signature = Base64Utils.encode(signBytes);
+
+    // Create an input for the transaction using the coinbase transaction hash
+    Input input = new Input(tx_out_hash_coinbase, 0, new ScriptInput(type, senderPublicKey, signature));
+
+    inputs.add(input);
+  }
+
+  /**
+   * Copied from Transaction data class in android frontend
+   *
+   * @return signature of all the outputs and inputs with the public key
+   */
+  public static String computeSigOutputsPairTxOutHashAndIndex(
+    List<Output> outputs, Map<String, Integer> inputsPairs) {
+    // input #1: tx_out_hash Value //input #1: tx_out_index Value
+    // input #2: tx_out_hash Value //input #2: tx_out_index Value ...
+    // TxOut #1: LaoCoin Value​​ //TxOut #1: script.type Value //TxOut #1: script.pubkey_hash Value
+    // TxOut #2: LaoCoin Value​​ //TxOut #2: script.type Value //TxOut #2: script.pubkey_hash
+    // Value...
+    List<String> sig = new ArrayList<>();
+
+    for (Map.Entry<String, Integer> current : inputsPairs.entrySet()) {
+      sig.add(current.getKey());
+      sig.add(String.valueOf(current.getValue()));
+    }
+
+    for (Output current : outputs) {
+      sig.add(String.valueOf(current.value));
+      sig.add(current.script.type);
+      sig.add(current.script.pubKeyHash);
+    }
+
+    return String.join("", sig.toArray(new String[0]));
+  }
+
+  /**
+   * Copied from Transaction data class in android frontend
+   *
+   * @return the transaction id computed based on the previous inputs and outputs.
+   */
+  public String computeId() {
+    // Make a list all the string in the transaction
+    List<String> collectTransaction = new ArrayList<>();
+    // Add them in lexicographic order
+
+    // Inputs
+    for (Input currentTxin : inputs) {
+      System.out.println("computing input: " + currentTxin);
```

Comment:
Is this some debug print or logs ?

It this is debug, I suggest you remove it.

If this is a log, you should probably be more verbose.

Reply:
It was a debug print, removed it :)